### ISSUE TITLE
Archive.extracted not working on 2016.11.6 Solaris 10 minion

### ISSUE THREAD

**Issue by Amacelia** [saltstack/salt#42863]
### Description of Issue/Question
On a Solaris 10 (2016.11.6)  minion, the below sls file using salt.state.archive.extracted is not working. It works on a Windows XP (2016.3.6) box to deploy software (using a slightly modified for windows paths sls file). So, the thought is it would work...so what is the issue?

### Setup
The file "salt://deploy_software.sls" contains:
```
deploy_software:
  archive.extracted:
    - name: '/var/tmp/<redacted>'
    - source: salt://<redacted>/files/deployment-file.tar.gz
    - archive_format: tar
    - if_missing: /var/tmp/<redacted>
```
Archive.extracted, in a similar sls file (with file paths changed for windows machines), works on a Salt 2016.3.6 Windows XP minion.

### Debug log:
error log after adding 'if-missing' option to sls file
```
[root@salt salt]# salt 'target-minion' state.sls deploy_software
target-minion:
----------
          ID: deploy-id
    Function: archive.extracted
        Name: <redacted path>
      Result: False
     Comment: Uncaught exception 'Error -3 while decompressing: invalid block type' when listing contents of /var/cache/salt/minion/files/base/_target_files_deployment-file.tar.gz. 
              Additional info follows:
              
              archive location:
                  /var/cache/salt/minion/files/base/_deployment-file.tar.gz
     Started: 22:48:08.341815
    Duration: 1098.587 ms
     Changes:   

Summary for target-minion
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   1.099 s
ERROR: Minions returned with non-zero exit code




-----------------------------------------------------------------------------------------------------
error before adding 'if_missing' option in the sls file ---->
--------------------------------------------------------
[root@salt salt]# salt-run jobs.print_job 20170809224603614274
20170809224603614274:
    ----------
    Arguments:
        - deploy_software
    Function:
        state.sls
    Result:
        ----------
        target-minion:
            ----------
            fun:
                state.sls
            id:
                target-minion
            jid:
                20170809224603614274
            master_id:
                salt_master
            retcode:
                2
            return:
                ----------
                archive_|-deploy_software_|-/var/tmp/<redacted>_|-extracted:
                    ----------
                    __id__:
                        deploy_software
                    __run_num__:
                        0
                    changes:
                        ----------
                    comment:
                        Uncaught exception 'Error -3 while decompressing: invalid block type' when listing contents of /var/cache/salt/minion/files/base/_target_files_deployment-file.tar.gz. 
                        Additional info follows:
                        
                        archive location:
                            /var/cache/salt/minion/files/base/_target_files_deployment-file.tar.gz.
                        
                        If the source archive is a tar archive compressed using a compression type not natively supported by the tar command, then setting the 'list_options' argument may allow the contents to be listed. Otherwise, if Salt is unable to determine the files/directories in the archive, 
                        the following workaround(s) would need to be used for this state to proceed (assuming the source file is a valid tar archive):
                        
                        - 'if_missing' must be set
                    duration:
                        119443.702
                    name:
                        /var/tmp/<redacted>
                    result:
                        False
                    start_time:
                        22:22:40.519202
            success:
                True
    StartTime:
        2017, Aug 09 22:46:03.614274
    Target:
        target-minion
    Target-type:
        glob
    User:
        root
```

### Versions Report
**Salt-master: ** CentOS 7 
Salt Version:
- Salt:  2016.11.6

Dependency Versions:
- cffi: 0.8.6
- cherrypy: Not Installed
- dateutil: 1.5
- gitdb: Not Installed
- gitpython: Not Installed
- ioflo: Not Installed
- Jinja2: 2.8
- libgit2: Not Installed
- libnacl: Not Installed
- M2Crypto: Not Installed
- Mako: 0.8.1
- msgpack-pure: Not Installed
- msgpack-python: 0.4.6
- mysql-python: Not Installed
- pycparser: 2.14
- pycrypto: 2.6.1
- pygit2: Not Installed
- Python: 2.7.5 (default, <date redacted>)
- python-gnupg: Not Installed
- PyYAML: 3.11
- PyZMQ: 15.3.0
- RAET: Not Installed
- smmap: Not Installed
- timelib: Not Installed
- Tornado: 4.2.1
- ZMQ: 4.1.4

System Versions:
- dist: centos 7.2.1511 Core
- machine: x86_64
- release: <number redacted>
- system: Linux
- version: CentOS Linux 7.2.1511 Core

**Salt-minion:** Solaris 10
Salt Version:
- Salt:  2016.11.6

Dependency Versions:
- cffi: 1.9.1
- cherrypy: Not Installed
- dateutil: 2.6.0
- gitdb: 0.6.4
- gitpython: 0.3.2 RC1
- ioflo: 1.6.6
- Jinja2: 2.8
- libgit2: Not Installed
- libnacl: 1.5.0
- M2Crypto: 0.25.1
- Mako: 1.0.6
- msgpack-pure: Not Installed
- msgpack-python: 0.4.8
- mysql-python: Not Installed
- pycparser: 2.17
- pycrypto: 2.6.1
- pygit2: Not Installed
- Python: 2.7.12 (default, <date redacted>)
- python-gnupg: 2.0.2
- PyYAML: 3.12
- PyZMQ: 16.0.2
- RAET: 0.6.5
- smmap: 0.9.0
- timelib: Not Installed
- Tornado: 4.4.2
- ZMQ: 4.0.7

System Versions:
- dist: 
- machine: <string redacted>
- release: <number redacted>
- system: SunOS
- version: Not Installed

**Comment by rickh563** [comment]
ZD-1675

**Comment by Amacelia** [comment]
Solved the problem. After running 'tar cvf' and 'gzip' on the file, I renamed the tar.gz file which caused archive.extracted to fail. 

To repeat the steps archive.extracted would take on the tar.gz file, I ran the 'gunzip' and 'tar xvf' steps to retrieve the original folder/files, and received a 'invalid compressed data--format violated' error on the 'gunzip' step. 

To solve, I ran 'gzip' and 'tar cvf' on the original folder/files and did not rename the tar.gz file and was able to run archive.extracted and have it execute successfully.
